### PR TITLE
Fix RMSNormGated in Zamba2

### PR DESCRIPTION
**Pull request by pglorio**
# What does this PR do?

This PR fixes `Zamba2RMSNormGated` to allow for `config.mamba_ngroups>1`. The Zamba2 7B checkpoints have `config.mamba_ngroups=2` so this change is necessary to have the correct forward pass.

I defined `Zamba2RMSNormGated` inside [modular_zamba2.py](https://github.com/huggingface/transformers/pull/35943/files#diff-94297873668a42780c28a94f5b612feef4b51307c11decbb9794f79a2b086f7aR73-R91) instead of importing it, as this differs from the definition in [modeling_mamba2.py](https://github.com/Zyphra/transformers_zamba/blob/cd304b5139198ac1c6ae822704347a7c645b6185/src/transformers/models/mamba2/modeling_mamba2.py#L210). The implementation in this PR is the torch version of the mamba-ssm implementation of the original mamba2 (used [here](https://github.com/state-spaces/mamba/blob/6b72c122713bb769cc82c6b8e6d019c53d27d6a1/mamba_ssm/modules/mamba2.py#L144) and torch implementation given [here](https://github.com/state-spaces/mamba/blob/60dadf2e0ee730ac337035d5533de10bc26e4847/mamba_ssm/ops/triton/layernorm_gated.py#L18)).

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [X] Did you write any new necessary tests?

## Who can review?

@ArthurZucker @Cyrilvallez

### CODE REVIEW

File: src/transformers/models/zamba2/modular_zamba2.py [hunk 1]
```diff
@@ -70,8 +70,25 @@
 logger = logging.get_logger(__name__)
 
 
-class Zamba2RMSNormGated(MambaRMSNormGated):
-    pass
+class Zamba2RMSNormGated(torch.nn.Module):
```
Comment (vasqu) [inline src/transformers/models/zamba2/modular_zamba2.py:73]: I think this will also affect the mamba2 code then (as codestral mamba also uses ngroups > 1) - so I'd be for implementing this in the mamba2 code and use modular then.

cc @molbap

Comment (pglorio) [reply src/transformers/models/zamba2/modular_zamba2.py:73]: @vasqu sounds good. Should I go ahead and update mamba2?

Comment (vasqu) [reply src/transformers/models/zamba2/modular_zamba2.py:73]: I think so, but as I'm no maintainer I leave the decision to the others 👀

**Review by ArthurZucker — COMMENTED** [review]
Hey! Just want to make sure, this is a fix and not a new feature specific to a model right?
@vasqu I know it is tempting to also add this for mamba2 but AFAIK this was not the original RMSNorm used then no?

TLDR; we don't make modeling changes unless it's bug fixes in general. If you have a new RMS norm it's a new model for use 😉

**Comment by vasqu** [comment]
@ArthurZucker I think it was an oversight in the original implementation for Mamba2 over here - @pglorio shows the relevant code snippets that `ngroups` is indeed used in the gated rms norm, e.g. [here](https://github.com/state-spaces/mamba/blob/6b72c122713bb769cc82c6b8e6d019c53d27d6a1/mamba_ssm/modules/mamba2.py#L144-L145) and [here](https://github.com/state-spaces/mamba/blob/60dadf2e0ee730ac337035d5533de10bc26e4847/mamba_ssm/ops/triton/layernorm_gated.py#L28-L36).

I can't estimate how it changes the model tho and if slow tests would need to be changed accordingly.

**Comment by vasqu** [comment]
I'll probably check some other time if logits match - maybe they indeed are equivalent 👀

**Review by ArthurZucker — APPROVED** [review]
Okay, let's only apply it to zamba for now!
